apps/api/src/simulation-run/simulation-run.service.ts

```diff
@@ -1179,7 +1202,7 @@ export class SimulationRunService {
                   '/' +
                   output.id +
                   '/' +
-                  curve.xDataGenerator.id +
+                  curve.xDataGenerator +
```

Comment:
This is because we are now storing everything as deserialized instead of just by id, correct?

Reply:
Correct.

Reply:
Everything is stored with object references serialized using ids. This is similar to the XML encoding of SED-ML. I added a method to deserialize these references.